src/js/cognitoConfig.js

```diff
@@ -45,3 +41,124 @@ exports.getToken = () => new Promise((resolve, reject) => {
+            return;
+        }
+        //TODO: wyświetlenie potwierdzenia jakiegoś
+    });
+}
+
+exports.getUserAttributes = () => {
+    const attributeList = [];
+    const cognitoUser = this.getUser();
+
+    if (cognitoUser != null) {
+        cognitoUser.getSession( (err, session) => {
+
+            if (err) {
+                alert(err.message || JSON.stringify(err));
+                return;
+            }
+
+            cognitoUser.getUserAttributes( (err, result) => {
+                if (err) {
+                    alert(err.message || JSON.stringify(err));
+                    return;
+                }
+                for (i = 0; i < result.length; i++) {
+                    const att = {
+                        Name: result[i].getName(),
+                        Value: result[i].getValue()
+                    };
+        
+                    attributeList.push(att);
+                }
+            });
+        });
+    }
+    return attributeList;
+};
+
+exports.updateUserAttributes = attributeList => {
+    const cognitoUser = this.getUser();
+
+    if (cognitoUser != null) {
+        cognitoUser.getSession( err => {
+
+            if (err) {
+                alert(err.message || JSON.stringify(err));
+                return;
+            }
+
+            cognitoUser.updateAttributes(attributeList, (err, result) => {
+                if (err) {
+                    alert(err.message || JSON.stringify(err));
+                    return;
+                }
+            });
+        });
+    }
+};
+
+exports.addUserAttribute = (attributeName, attributeValue) => {
+    const attributeList = this.getUserAttributes();
+    const attributeData = {
+        Name: attributeName,
+        Value: attributeValue,
+    };
+    const attribute = new AmazonCognitoIdentity.CognitoUserAttribute(attributeData);
+    attributeList.push(attribute);
+
+    this.updateUserAttributes(attributeList);
+}
+
+exports.addCustomUserAttribute = (attributeName, attributeValue) => {
+    this.addUserAttribute('custom:' + attributeName, attributeValue);
+}
+
+exports.checkUserAttribute = attributeName => {
+    const attributeList = [];
+    const cognitoUser = this.getUser();
+
+    if (cognitoUser != null) {
+        cognitoUser.getSession( (err, session) => {
+
+            if (err) {
+                alert(err.message || JSON.stringify(err));
+                return;
+            }
+
+            cognitoUser.getUserAttributes( (err, result) => {
+                if (err) {
+                    alert(err.message || JSON.stringify(err));
+                    return;
+                }
+                for (i = 0; i < result.length; i++) {
+                    if(result[i].getName() == attributeName) {
+                        attributeList.push(result[i].getValue());
+                    }
+                }
+            });
+        });
+    }
+    return attributeList; //tutaj coś jest nie tak; nie mogę zwrócić wartości

```

Comment:
Tutaj przydałaby się czyjaś pomoc, bo nie potrafię wyłuskać wartości tego atrybutu w żaden sposób...
Przekazywane jest np. coś takiego: [{"piruci88"}] i zwykłe zrobienie attributeList[0] nie działa.

Reply:
Co to jest [{"piruci88"}]? Nie ma czegoś takiego w js xD

Reply:
Wejdź na chwile na brancha i sam zobacz co to jest xD

Reply:
Chciałem, ale chyba nigdzie nie ma tego wywołanego